internal/helm/service.go

```diff
@@ -45,11 +45,37 @@ type Repository struct {
 	TlsSecretID string `json:"tlsSecretId,omitempty"`
 }
 
+// Options struct holding directives for driving helm operations (similar to command line flags)
+// extend this as required eventually build a mor sophisticated solution for it
+type Options struct {
+	Namespace    string                 `json:"namespace,omitempty"`
+	DryRun       bool                   `json:"dryRun,omitempty"`
+	GenerateName bool                   `json:"generateName,omitempty"`
+	Wait         bool                   `json:"wait,omitempty"`
+	Timeout      int64                  `json:"timeout,omitempty"`
+	OdPcts       map[string]interface{} `json:"odPcts,omitempty"`
+	ReuseValues  bool                   `json:"reuseValues,omitempty"`
+	Optionals    map[string]interface{}
+}
+
 // +kit:endpoint:errorStrategy=service
 // +testify:mock:testOnly=true
 
 // Service manages Helm chart repositories.
 type Service interface {
+	// helm repository management operations
+	repository
+
+	// release management operations
+	releaser
+
+	// chart related operations
+	charter
```

Comment:
Interesting approach....I'm not sure I like it though. 😛 

The interface is harder to read this way.

You can split the implementation of the service into categories (eg. `service_release.go`).

Reply:
thx, will do it in a later PR